trio/_core/_io_kqueue.py

```diff
@@ -19,7 +19,7 @@ class _KqueueStatistics:
 
 @attr.s(slots=True, eq=False)
 class KqueueIOManager:
-    _kqueue = attr.ib(factory=select.kqueue)
+    _kqueue = attr.ib(factory=select.kqueue)  # type: ignore
```

Comment:
This works because we require Linux when running mypy. But if I don't specify the platform, I get the following errors on macOS:

```
trio/_core/_io_kqueue.py:22: error: unused 'type: ignore' comment
trio/_core/_io_epoll.py:186: error: Module has no attribute "epoll"; maybe "poll"?
```

I plan to fix that in a further pull request by adding `assert sys.platform.startswith("freebsd") or sys.platform.startswith("darwin")` at the top of the file and running mypy on the various platforms we support.